include/aws/http/private/h2_connection.h

```diff
@@ -75,11 +76,13 @@ struct aws_h2_connection {
          * When queue is empty, then we send DATA frames from the outgoing_streams_list */
         struct aws_linked_list outgoing_frames_queue;
 
-        /* Maps stream-id to aws_h2_stream_closed_when.
-         * Contains data about streams that were recently closed by this end (sent RST_STREAM frame or END_STREAM flag),
-         * but might still receive frames that remote peer sent before learning that the stream was closed.
-         * Entries are removed after a period of time. */
-        struct aws_hash_table closed_streams_where_frames_might_trickle_in;
+        /* LRU cache for closed stream, key: steam-id, value: aws_h2_stream_closed_detail.
+         * Contains data about streams that were recently closed.
+         * Entries are removed after a period of time. (AWS_H2_IGNORE_TIME) */
+        struct aws_lru_cache closed_streams;
+
+        /* Max size for the closed stream cache */
+        size_t max_closed_stream_cache_size;
```

Comment:
I don't see a reason to bloat the memory by 8 bytes. This just gets passed in to the init for the lru cache.